src/cli/commands/start.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @module cli/commands
+ */
+import fs from "fs";
+import {CliCommand} from "./interface";
+import {CommanderStatic} from "commander";
+import {ILogger, LogLevel, WinstonLogger} from "../../logger";
+import BeaconNode, {BeaconNodeCtx} from "../../node";
+import {RpcClientOverInstance, RpcClientOverWs} from "../../validator/rpc";
+import {ValidatorCtx} from "../../validator/types";
+import {Keypair} from "@chainsafe/bls-js/lib/keypair";
+import Validator from "../../validator";
+import defaults from "../../validator/defaults";
+import {BeaconApi} from "../../rpc/api/beacon";
+import {BeaconDB, ValidatorDB} from "../../db/api";
+import {BeaconChain} from "../../chain";
+import {ValidatorApi} from "../../rpc/api/validator";
+import {PrivateKey} from "@chainsafe/bls-js/lib/privateKey";
+import Keystore from "../../validator/keystore";
+import {promptPassword} from "../../util/io";
+import {BeaconNodeCommand} from "./beacon";
+import {LevelDbController} from "../../db/controller";
+
+interface IStartCommandOptions {
+  db: string;
+  depositContract: string;
+  eth1RpcUrl: string;
+  rpc: string;
+  configFile: string;
+  key: string;
+  dbValidator: string;
+  loggingLevel: string;
+}
+
+export class StartCommand implements CliCommand {
+
+  public register(commander: CommanderStatic): void {
+    const logger: ILogger = new WinstonLogger();
+
+    commander
+      .command("start")
+      .description("Start lodestar node and bundled validator")
+      .option("--db [db_path]", "Path to Beacon node database")
+      .option("-dc, --depositContract [address]", "Address of deposit contract")
+      .option("-eth1, --eth1RpcUrl [url]", "Url to eth1 rpc node")
+      .option("--rpc [api]", "Exposes the selected RPC api, must be comma separated")
+      .option("-c, --configFile [config_file]", "Config file path")
+
+      .option("-k, --key [key]", "Private key of the validator")
+      .option("--dbValidator [db_path]", "Path to the validator database")
+      .option(`-l, --loggingLevel [${Object.values(LogLevel).join("|")}]`, "Logging level")
+      .action(async (options) => {
+        // library is not awaiting this method so don't allow error propagation
+        // (unhandled promise rejections)
+        try {
+          let nodeCommand = new BeaconNodeCommand();
+          await nodeCommand.action(options, logger);
+          await this.action(options, nodeCommand.node, logger);
+        } catch (e) {
+          logger.error(e.message + '\n' + e.stack);
+        }
+      });
+  }
+
+  public async action(options: IStartCommandOptions, node: BeaconNode, logger: ILogger): Promise<void> {
+    if (options.loggingLevel) {
+      logger.setLogLevel(LogLevel[options.loggingLevel]);
+    }
+
+    let dbName: string;
+    if (options.dbValidator) {
+      dbName = options.dbValidator;
+    } else {
+      dbName = defaults.db.name;
+    }
+    let db = new ValidatorDB({
```

Comment:
I don't think logic for database and api initialization should be in cli command code. You should add validator initialization code to file in `src/node/validator.ts`. and if starting bundled just add ifology to `src/node/index.ts` to invoke  `src/node/validator.ts`

Reply:
I will move the  database and api initialization from the cli command code. But have you meant `src/validator/validator.ts` instead of `src/node/validator.ts`?

Reply:
I'm up for validator as service. @GregTheGreek @wemeetagain ?

Reply:
I think we should have a separate validator "node" that runs separately from the beacon node. This way we can keep our node/index.ts beacon-specific, have more reuse, and less special-casing in our nodes.
`initValidator` or whatever we call it can handle configuring the validator node to use the beacon node's apis vs a rpc url.
Then we will simply have a "normal" beacon node running along side another "normal" validator node. The only difference is that they'll both be in the same process and the validator's rpc will be configured differently. This pattern would also make it easier to put the validator and beacon nodes in different workers in the future.
See https://github.com/ChainSafe/lodestar/pull/253/files#r296780603

Reply:
yep. I dont like the idea of running them on the same thread, they need to be independent because in theory the validator should connect to multiple beacon nodes. should the local one fail it will kill the validator one (maybe) by accident and slash funds (maybe)

Reply:
I'd almost veto having a bundled one at this point until we figure out a solution for it